src/components/modebar/buttons.js

```diff
@@ -253,13 +253,6 @@ function handleCartesian(gd, ev) {
         if(astr === 'hovermode' && (val === 'x' || val === 'y')) {
             val = fullLayout._isHoriz ? 'y' : 'x';
             button.setAttribute('data-val', val);
-        } else if(astr === 'hovermode' && val === 'closest') {
-            for(i = 0; i < axList.length; i++) {
-                ax = axList[i];
-                if(allSpikesEnabled === 'on' && !ax.showspikes) {
-                    allSpikesEnabled = 'off';
-                }
-            }
```

Comment:
Some history here:

- Cartesian spikes were first introduced in https://github.com/plotly/plotly.js/pull/1461
- back then, they only worked with `hovermode: 'closest'`
- this particular block got added in https://github.com/rpaskowitz/plotly.js/commit/762b54ab3a30189c14d4833dc4eebd11552bb158 w/o tests
- https://github.com/plotly/plotly.js/pull/2247 made spikes work regardless of `hovermode`, but didn't :hocho: this block